### PR TITLE
SuperscalarEventSets check width

### PR DESCRIPTION
**Pull request by johningalls-sifive**
**Related issue**: add the `require` from https://github.com/chipsalliance/rocket-chip/pull/2125 onto https://github.com/chipsalliance/rocket-chip/pull/2337

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**
require that `SuperscalarEventSets` events fit within mask, same as regular `EventSets`

### CODE REVIEW

File: src/main/scala/rocket/Events.scala [hunk 1]
```diff
@@ -28,14 +28,15 @@ class EventSet(gate: (UInt, UInt) => Bool, val events: Seq[(String, () => Bool)]
...
+    val maskMask = ((BigInt(1) << eventSets.map(_.size).max) - 1) << maxEventSetIdBits
     eventSel & (setMask | maskMask).U
   }
 
   private def decode(counter: UInt): (UInt, UInt) = {
-    require(eventSets.size <= (1 << eventSetIdBits))
-    (counter(log2Ceil(eventSets.size)-1, 0), counter >> eventSetIdBits)
+    require(eventSets.size <= (1 << maxEventSetIdBits))
+    require(eventSetIdBits > 0)
+    (counter(eventSetIdBits-1, 0), counter >> maxEventSetIdBits)
```
Comment (leikou01) [inline src/main/scala/rocket/Events.scala:39]: on the high level, what is the purpose of this change? 
differentiate the actually eventSet and maxEventSet? and maxEventSet will be limited by the architecture?

Comment (johningalls-sifive) [reply src/main/scala/rocket/Events.scala:39]: just making the code the same for both `EventSet` and `SuperscalarEventSet`
